README.md

```diff
@@ -67,3 +67,120 @@ teracy-dev:
         branch: develop
       require_version: ">= 0.4.0-SNAPSHOT"
 ```
+
+## Supported configuration
+These are the supported configuration in teracy-dev-core: `variables`, `vagrant`, `default`, `node`.
+
+
+### variables
+`variables` which is the environment variables must be defined to use `teracy-dev`. All variables
+ setting should be defined in the `variables` block.
+
+ The following is the  default configuration:
+
+```yaml
+# vars must be defined to be used
+# format:
+# key: ${ENV_VAR}
+# key: ${ENV_VAR-:default}
+# key: value
+# and "%{key}" can be used for settings values if available
+variables:
+  node_name_prefix: ${NODE_NAME_PREFIX:-node}
+  node_hostname_prefix: ${NODE_HOSTNAME_PREFIX:-node}
+  node_domain_affix: ${NODE_DOMAIN_AFFIX:-local}
+  # some_value: ${MY_VAR:-value}-affix
+```
+In which:
+
+- `node_name_prefix` param: its value is `NODE_NAME_PREFIX`. If `NODE_NAME_PREFIX` is not defined, the value
+should be the default `node`.
+- `node_hostname_prefix` param: its value is `NODE_HOSTNAME_PREFIX`. If `NODE_HOSTNAME_PREFIX` is not defined, the value should be the default `node`.
+- `node_domain_affix` param: its value is `NODE_DOMAIN_AFFIX`. If `NODE_DOMAIN_AFFIX` is not defined, the value should be the default `local`.
+
+
+### vagrant
+
+The `vagrant` configuration is used for the vagrant settings. Vagrant plugins are actually
+rubygems so you can declare to use the common configuration level of any rubygems.
+
+The following is the default configuration:
+
+```yaml
+  # vagrant related settings
+vagrant:
+  # vagrant plugins are actually rubygems so you can declare to use any rubygems
+  # common config level
+  plugins: []
+    # - _id: "0"
+    #   name: vagrant-gatling-rsync
+    #   version: "" # empty means latest by default
+    #   state: installed # uninstalled
+    #   enabled: true
+    #   config_key: gatling
+    #   options:
+    #     latency: 0.5
+    #     time_format: "%H:%M:%S"
+    #     rsync_on_startup: true
+```
+
+### default
+
+The `default` settings which is used for all nodes and should be defined in the `default` block. It
```

Comment:
which will be overridden by each `nodes`, so any config which is shared among nodes, we should use `default`